git.go

```diff
@@ -40,79 +72,40 @@ func openOrCloneRepo() error {
 }
```

Comment:
There is an issue in this function, if at any point there is an error triggered during the clone routine, it will leave the `repo.path` folder dirty with an unusable `.git` folder in it. This causes `git2etcd` to first return with a `FATAL` error while following runs of `git2etcd` will just print a warning that it can't update as it does not go through the clone step.

This can be extremely frustrating for a user as a first misconfiguration might still break any succeeding run with a valid config.

An example of this issue would be the one I stumbled upon while testing the ssh clone routine.

- First run
```
WARN[0000] Couldn't find repo locally, trying to clone it  error="repository does not exist"
INFO[0000] Cloning repo                                  branch=my-branch path=data/ url="git@github.com:0rax/my-private-repo.git"
FATA[0000] Couldn't find repo or clone it                error="knownhosts: /Users/orax/.ssh/known_hosts:4: illegal base64 data at input byte 376"
```

- Second run
```
WARN[0001] Couldn't sync repo                            error="Couldn't pull: error creating SSH agent: \"SSH agent requested but SSH_AUTH_SOCK not-specified\""
```

Reply:
Moving this comment as a separate issue as it does not seem trivial and is not really part of the scope of this migration.

Looking at the code, this was also part of the legacy code but was seen due to a typo in my `~/.ssh/known_hosts`. Let's not wait for this needed library migration and work on this issue separatly.